### PR TITLE
chore: deprecate edx sphinx theme

### LINKED ISSUE #23: Deprecate edx-sphinx-theme
We need to remove `edx-sphinx-theme` and use `sphinx-book-theme` in place of that.

### PR DESCRIPTION
**Pull request by huniafatima-99**
This PR deprecates the `edx-sphimx-theme` and replaces it with `sphinx-book-theme`
Issue Link: https://github.com/edx/edx-arch-experiments/issues/339

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -172,16 +171,12 @@ def get_version(*file_paths):
...
 # a list of builtin themes.
 
-html_theme = 'edx_theme'
+html_theme = 'sphinx_book_theme'
 
 # Theme options are theme-specific and customize the look and feel of a theme
 # further.  For a list of options available for each theme, see the
 # documentation.
 #
-# html_theme_options = {}
```
Comment (iamsobanjaved) [inline docs/conf.py:181]: is this line removal necessary to work, if yes then related comment should also be removed.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -15,24 +15,24 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ['3.8']
-        toxenv: [quality, docs, 'py38']
+        python-version: ['3.9']
+        toxenv: [quality, docs, 'py39']
```
Comment (iamsobanjaved) [inline .github/workflows/ci.yml]: We should upgrade the Python version to 3.11 as this is version currently used in most IDAs

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -15,24 +15,24 @@ jobs:
...
         env:
           TOXENV: ${{ matrix.toxenv }}
         run: tox
-      - name: Run Coverage
-        if: matrix.toxenv == 'py38'
-        uses: codecov/codecov-action@v3
-        with:
-          flags: unittests
-          fail_ci_if_error: true
+#      - name: Run Coverage
```
Comment (iamsobanjaved) [inline .github/workflows/ci.yml]: Was codecov failing here?

File: requirements/doc.txt [hunk 4]
```diff
@@ -1,17 +1,23 @@
 #
-# This file is autogenerated by pip-compile with Python 3.8
+# This file is autogenerated by pip-compile with Python 3.9
 # by the following command:
 #
-#    make upgrade
+#    pip-compile --output-file=requirements/doc.txt requirements/doc.in
```
Comment (iamsobanjaved) [inline requirements/doc.txt]: We should not compile a single file only as it causes inconsistencies in requirements. Instead, we can upgrade a single requirement by temporarily removing the `--upgrade` flag from `pip compile` commands within the make upgrade target.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (huniafatima-99) [reply .github/workflows/ci.yml]: yes codecov is failing here.

**Review by UsamaSadiq — DISMISSED** [review]
Can we upgrade it to use `Python 3.12` instead?

**Review by iamsobanjaved — APPROVED** [review]

**Review by iloveagent57 — APPROVED** [review]

**Pull request merged**